src/qibo/backends/__init__.py

```diff
@@ -8,66 +9,61 @@
 from qibo.backends.tensorflow import TensorflowBackend
 from qibo.config import log, raise_error
 
-
-def construct_backend(backend, **kwargs):
-    if backend == "qibojit":
-        from qibojit.backends import CupyBackend, CuQuantumBackend, NumbaBackend
-
-        platform = kwargs.get("platform")
-        if platform == "cupy":  # pragma: no cover
-            return CupyBackend()
-        elif platform == "cuquantum":  # pragma: no cover
-            return CuQuantumBackend()
-        elif platform == "numba":
-            return NumbaBackend()
-        else:  # pragma: no cover
+QIBO_NATIVE_BACKENDS = ("numpy", "tensorflow", "pytorch")
+QIBO_NON_NATIVE_BACKENDS = ("qibojit", "qibolab", "qibo-cloud-backends", "qibotn")
+
+
+class MetaBackend:
+    """Meta-backend class which takes care of loading the qibo backends."""
+
+    @staticmethod
+    def load(backend: str, **kwargs) -> Backend:
+        """Loads the backend.
+
+        Args:
+            backend (str): Name of the backend to load.
+            kwargs (dict): Additional arguments for the non-native qibo backends.
+        Returns:
+            qibo.backends.abstract.Backend: The loaded backend.
+        """
+
+        if backend == "numpy":
+            return NumpyBackend()
+        elif backend == "tensorflow":
+            return TensorflowBackend()
+        elif backend == "pytorch":
+            return PyTorchBackend()
+        elif backend == "clifford":
+            engine = kwargs.pop("platform", None)
+            kwargs["engine"] = engine
+            return CliffordBackend(**kwargs)
+        elif backend in QIBO_NON_NATIVE_BACKENDS:
+            module = import_module(backend.replace("-", "_"))
+            return getattr(module, "MetaBackend").load(**kwargs)
```

Comment:
I'm not sure whether I'd nest that here.

Instead, we could make a separate function, and the Qibo `MetaBackend` could be just one like anyone else, providing a nice example of how to implement one.

Eventually, we could discover backends with some runtime mechanism, without relying on something like
https://github.com/qiboteam/qibo/blob/f43a6a1a7eea95c0da1ecb65a9c6d346ecfe0a0a/src/qibo/backends/__init__.py#L13
(e.g. by having installed additional packages prefixed with `qibo-backend-` or something like that).
But I'd keep it simple for this PR, just properly lay the ground.